src/plone/api/user.py

```diff
@@ -298,11 +299,10 @@ def get_permissions(username=None, user=None, obj=None):
 
     result = {}
     with context:
-        portal_membership = portal.get_tool('portal_membership')
-        permissions = (permission[0] for permission in getPermissions())
-        for permission in permissions:
+        for record in getPermissions():
+            permission = record[0]
             result[permission] = bool(
-                portal_membership.checkPermission(permission, obj),
+                getSecurityManager().checkPermission(permission, obj),
```

Comment:
If you are really interested in performance I would define getSecurityManager() out of the for cycle.
```
sm = getSecurityManager()
permissions = (record[0] for record in getPrermissions())
result = {permission: bool(sm.checkPermission(permission, obj) for permission in permissions}
```

Reply:
indeed!